cmd/capibmadm/cmd/vpc/key/list.go

```diff
@@ -0,0 +1,146 @@
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/printer"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+	pagingUtil "sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/utils"
+)
+
+var resourceGroupName string
+
+// ListCommand vpc key list command.
+func ListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List VPC key",
+		Example: `
+# List key in VPC
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm vpc key list --region <region> --resource-group-name <resource-group-name>`,
+	}
+
+	options.AddCommonFlags(cmd)
+	options.AddVPCCommonFlags(cmd)
+
+	cmd.Flags().StringVar(&resourceGroupName, "resource-group-name", resourceGroupName, "IBM cloud resource group name")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := listKeys(cmd.Context(), resourceGroupName); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func listKeys(ctx context.Context, resourceGroupName string) error {
+	v1, err := utils.CreateVPCService(options.GlobalOptions.VPCRegion)
+	if err != nil {
+		return err
+	}
+
+	accountID, err := utils.GetAccountID(ctx, utils.GetIAMAuth())
+	if err != nil {
+		return err
+	}
+
+	if resourceGroupName != "" {
+		_, err = utils.GetResourceGroupID(ctx, resourceGroupName, accountID)
+		if err != nil {
+			return err
+		}
+	}
+
+	var keyNesList []*vpcv1.KeyCollection
+	f := func(start string) (bool, string, error) {
+		var listKeyOpt vpcv1.ListKeysOptions
+
+		if start != "" {
+			listKeyOpt.Start = &start
+		}
+
+		keyL, _, err := v1.ListKeysWithContext(ctx, &listKeyOpt)
+		if err != nil {
+			return false, "", err
+		}
+		keyNesList = append(keyNesList, keyL)
+
+		if keyL.Next != nil && *keyL.Next.Href != "" {
+			return false, *keyL.Next.Href, nil
+		}
+
+		return true, "", nil
+	}
+
+	if err = pagingUtil.PagingHelper(f); err != nil {
+		return err
+	}
+
+	return display(keyNesList)
+}
+
+func display(keyNesList []*vpcv1.KeyCollection) error {
+	var keyListToDisplay List
+	for _, keyL := range keyNesList {
+		for _, key := range keyL.Keys {
+			keyToAppend := Key{
+				CreatedAt:     utils.DereferencePointer(key.CreatedAt).(strfmt.DateTime),
+				ID:            utils.DereferencePointer(key.ID).(string),
+				Name:          utils.DereferencePointer(key.Name).(string),
+				Type:          utils.DereferencePointer(key.Type).(string),
+				Length:        utils.DereferencePointer(key.Length).(int64),
+				FingerPrint:   utils.DereferencePointer(key.Fingerprint).(string),
+				ResourceGroup: utils.DereferencePointer(key.ResourceGroup.Name).(string),
+			}
+
+			if key.ResourceGroup != nil {
+				keyToAppend.ResourceGroup = utils.DereferencePointer(key.ResourceGroup.Name).(string)
+			}
+
+			keyListToDisplay = append(keyListToDisplay, keyToAppend)
+		}
+	}
+
+	p, err := printer.New(options.GlobalOptions.Output, os.Stdout)
+
```

Comment:
nit: generally err check can be done in next line itself after assignment, new line may not be required.